crates/bfv-helpers/src/client.rs

```diff
@@ -103,12 +108,26 @@ where
         .try_encrypt_extended(&plaintext, &mut thread_rng())
         .map_err(|e| anyhow!("Error encrypting data: {}", e))?;
 
+    // Reconstruct e1_rns in mod Q.
+    let mut e1_power = e1_rns.clone();
+    e1_power.change_representation(Representation::PowerBasis);
+
+    // This conversion internally calls lift for each coefficient
+    // to make them in mod Q.
+    let e1_mod_q: Vec<BigUint> = Vec::<BigUint>::from(&e1_power);
+
+    // Then, make it a polynomial in mod Q.
+    let ctx = params.ctx()[0].clone();
+    let e1_bigints: Vec<BigInt> = e1_mod_q.iter().map(|c| c.to_bigint().unwrap()).collect();
+    let e1 = (*Poly::from_bigints(&e1_bigints, &ctx)?).clone();
+
```

Comment:
@0xjei , I really don't like duplicating code :D it would be better to move this to zkfhe-generator, even a fast solution